### PR TITLE
[FIX] User can block himself

### LINKED ISSUE #1278: [BUG] User can block himself
Do not let the user block receiving his own messages!

### PR DESCRIPTION
**Pull request by tjnet**
@RocketChat/ios

Closes #1278 

I'm new to this. Please review in your free time.

#### Where should the reviewer start?

Auth.swift

#### How should this be manually tested?

try to block my own message

#### Questions:

- naming is good?
- Should I change another file?

### CODE REVIEW

**Comment by Hrayr2112** [comment]
Next time when working on an issue keep updated please

**Comment by tjnet** [comment]
thank you for your feedback!
should I resubmit?

**Comment by Hrayr2112** [comment]
I am not a mentor. I am an another developer, like you.

пн, 19 февр. 2018 г. в 3:08, tjnet <notifications@github.com>:

File: Rocket.Chat/Controllers/Chat/ChatControllerMessageActions.swift [hunk 1]
```diff
@@ -36,15 +36,16 @@ extension ChatViewController {
...
-        alert.addAction(UIAlertAction(title: localized("chat.message.actions.block"), style: .default, handler: { [weak self] (_) in
-            guard let user = message.user else { return }
-
-            DispatchQueue.main.async {
-                MessageManager.blockMessagesFrom(user, completion: {
-                    self?.updateSubscriptionInfo()
-                })
-            }
-        }))
+        if AuthManager.isAuthenticated()?.canBlockMessage(message) == .allowed {
```
Comment (cardoso) [inline Rocket.Chat/Controllers/Chat/ChatControllerMessageActions.swift]: 👍

File: Rocket.Chat/Models/Auth.swift [hunk 2]
```diff
@@ -145,3 +145,25 @@ extension Auth {
         return .allowed
     }
 }
+
+extension Auth {
+    enum CanBlockMessageResult {
+        case allowed
+        case notActionable
```
Comment (cardoso) [inline Rocket.Chat/Models/Auth.swift]: Great! 😃

File: Rocket.ChatTests/Models/AuthSpec.swift [hunk 3]
```diff
@@ -273,4 +273,48 @@ class AuthSpec: XCTestCase, RealmTestCase {
         XCTAssert(auth.canEditMessage(message) == .allowed)
     }
 
+    func testCanBlockMessage() {
+        let realm = testRealm()
```
Comment (cardoso) [inline Rocket.ChatTests/Models/AuthSpec.swift]: 👏

**Pull request merged**